cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaSetOperator.java

```diff
@@ -29,24 +28,17 @@ public KafkaSetOperator(Vertx vertx, KubernetesClient client, long operationTime
     }
 
     @Override
-    protected Future<ReconcileResult<Boolean>> internalPatch(String namespace, String name, StatefulSet current, StatefulSet desired) {
+    protected boolean shouldIncrementGeneration(StatefulSet current, StatefulSet desired) {
         StatefulSetDiff diff = new StatefulSetDiff(current, desired);
         if (diff.changesVolumeClaimTemplates()) {
             log.warn("Changing Kafka storage type or size is not possible. The changes will be ignored.");
             diff = revertStorageChanges(current, desired);
         }
-        if (diff.isEmpty()) {
-            return Future.succeededFuture(ReconcileResult.noop());
-        } else {
-            boolean different = needsRollingUpdate(diff);
-            return super.internalPatch(namespace, name, current, desired).map(r -> {
-                if (r instanceof ReconcileResult.Patched) {
-                    return ReconcileResult.patched(different);
-                } else {
-                    return r;
-                }
-            });
+        if (diff.changesVolumeClaimTemplates()) {
```

Comment:
Why do we have now this block twice here with one of them reffering to _Zookeeper_? Seems like a bug.

Reply:
Good spot @scholzj